test/test.js

```diff
@@ -21172,6 +21172,47 @@
 
   /*--------------------------------------------------------------------------*/
 
+  QUnit.module('lodash.asArray');
+
+  (function() {
+    var wrapCases = [null, void 0, 1, '', 'ab', 0, {a: 1}, Object('xyz'), NaN];
+
+    QUnit.test('should wrap non array items in an array', function(assert) {
+      assert.expect(wrapCases.length * 2);
+
+      _.each(wrapCases, function(val) {
+        var result = _.asArray(val);
+        assert.deepEqual(result, [val], 'for value ' + val);
+        assert.strictEqual(result[0], val, 'should not copy the value');
```

Comment:
test262's `sameValue` handles NaN comparison, it makes me want to extend QUnit to have a similar assertion, including a `sameValueZero` (where it compares `-0` as equal to `+0`).

Would it be useful on LoDash?